Fix issue #505 removing trailing \r from lines in Windows text files

tuantrieu opened the pull request:
Window uses \r\n to split lines. so, for text files from Windows, when
\n is used as delimiter, \r is still appear at the end of lines.

The code checks to see if \n is used as delimiter and end of a line is
\r, then \r is removed from the line

tuantrieu committed:
Fix issue #505 removing trailing \r from lines in Windows text files

Window uses \r\n to split lines. so, for text files from Windows, when
\n is used as delimiter, \r is still appear at the end of lines.

The code checks to see if \n is used as delimiter and end of a line is
\r, then \r is removed from the line

rmetzger commented:
Hi,
thank you for your pull request.
your code is breaking some of our test cases. Have a look at the output of travis: https://travis-ci.org/stratosphere/stratosphere/builds/20784188
You can run the tests locally as well using `mvn verify`. You can also start individual JUnit Tests from Eclipse/IntelliJ

tuantrieu commented:
Thank, I didn't know there are unit test cases. I fixed it and running test cases.

I notice that there are 2TextInputFormat.java files in package eu.stratosphere.api.java.record.io and package eu.stratosphere.api.java.io.

Thus, I have to make 2 identical changes in the 2 files.

tuantrieu committed:
Fix issue #505 removing trailing \r from lines in Windows text files

Window uses \r\n to split lines. so, for text files from Windows, when
\n is used as delimiter, \r is still appear at the end of lines.

The code checks to see if \n is used as delimiter and end of a line is
\r, then \r is removed from the line

rmetzger commented:
Duplicate of https://github.com/stratosphere/stratosphere/pull/582
